src/web/js/ide.js

```diff
@@ -79,7 +106,28 @@ const runtimeApi = {
              (TODO: return richer values for error returns)
   */
   execute(bytecode) {
-    return new Promise((resolve, reject) => reject(new Error("Not Implemented")));
+    var runtime = window.CPOIDEHooks.runtime;
```

Comment:
Yup, good idea. That also avoids my motivation for having methods, since it gives us a scope to define local helpers.